simulator-ui/tests/home.spec.ts

```diff
@@ -0,0 +1,169 @@
+  await mockBackendResponse(page, '**/api/test-results/count-by-status', {
+    "successful": nbOfSuccessfulTests,
+    "failed": nbOfFailedTests,
+    "total": nbOfTotalTests
+  });
+  await page.getByTestId('refreshListButton').click();
+
+  await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, nbOfTotalTests, nbOfSuccessfulTests, nbOfFailedTests);
+})
+
+test('should have updated total, successful, failed tabs after refresh button clicked negative test with false total', async ({page}) => {
+  await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, nbOfTotalTests, nbOfSuccessfulTests, nbOfFailedTests);
+  nbOfFailedTests -= 10; // so the total will be wrong!
+  const newCorrectTotal: number = nbOfTotalTests - 10;
+  await mockBackendResponse(page, '**/api/test-results/count-by-status', {
+    "successful": nbOfSuccessfulTests,
+    "failed": nbOfFailedTests,
+    "total": nbOfTotalTests
+  })
+
+  await page.getByTestId('refreshListButton').click();
+
+  await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, newCorrectTotal, nbOfSuccessfulTests, nbOfFailedTests);
+})
+
+test('should have same total, successful, failed tabs after cancel deletion via close-Button and cancel-Button', async ({page}) => {
+  await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, nbOfTotalTests, nbOfSuccessfulTests, nbOfFailedTests);
+  const closeButtons = ['testResultDeleteDialogCloseButton', 'testResultDeleteDialogCancelButton'];
+
+  for (const button of closeButtons) {
+    await page.getByTestId('resetButton').click();
+    await expect(page.getByTestId('testResultDeleteDialogHeading')).toBeVisible();
+    await page.getByTestId(button).click();
+    // Do I have to assert that API was NOT called here? --> isn't that covered by unit test?
+    await expect(page.getByTestId('testResultDeleteDialogHeading')).toBeHidden();
+
+    await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, nbOfTotalTests, nbOfSuccessfulTests, nbOfFailedTests);
+  }
+})
+
+test('should have reset total, successful, failed tabs after confirmed deletion with (200, OK) response', async ({page}) => {
+  await checkIfSummaryTabsAreDisplayingRightNumbersFunction(page, nbOfTotalTests, nbOfSuccessfulTests, nbOfFailedTests);
+  await page.getByTestId('resetButton').click();
+  await expect(page.getByTestId('testResultDeleteDialogHeading')).toBeVisible();
+
+  nbOfSuccessfulTests = 0;
+  nbOfFailedTests = 0;
+  nbOfTotalTests = 0;
+  await mockBackendResponse(page, '**/api/test-results/count-by-status', {
+    "successful": nbOfSuccessfulTests,
+    "failed": nbOfFailedTests,
+    "total": nbOfTotalTests
+  })
+  //are these promises too much because api-response-handling should be already covered in unit-tests?
+  //if they stay: Should I make an external method for the deleteReq/RespPromises for readability? --> no because it isn't reused?
+  const deleteRequestPromise = page.waitForRequest(request =>
```

Comment:
are these promises too much because api-response-handling should be already covered in unit-tests?

Reply:
no